From 5cef084ec56ee326d3b169c2c83950a017ae02f6 Mon Sep 17 00:00:00 2001
From: Duncan Ogilvie <mr.exodia.tpodt@gmail.com>
Date: Wed, 29 Oct 2025 00:52:58 +0100
Subject: [PATCH] 8597: Fix narrowing conversion warning

error : constant expression evaluates to -1 which cannot be narrowed to type 'uintb' (aka 'unsigned long long') [-Wc++11-narrowing]
---
 Ghidra/Features/Decompiler/src/decompile/cpp/address.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/address.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/address.cc
index 9e8039d486e..a939961c371 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/address.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/address.cc
@@ -630,8 +630,8 @@ void RangeList::decode(Decoder &decoder)
 #ifdef UINTB4
 uintb uintbmasks[9] = { 0, 0xff, 0xffff, 0xffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff };
 #else
-uintb uintbmasks[9] = { 0, 0xff, 0xffff, 0xffffff, 0xffffffff, 0xffffffffffLL,
-			0xffffffffffffLL, 0xffffffffffffffLL, 0xffffffffffffffffLL };
+uintb uintbmasks[9] = { 0, 0xff, 0xffff, 0xffffff, 0xffffffff, 0xffffffffffULL,
+			0xffffffffffffULL, 0xffffffffffffffULL, 0xffffffffffffffffULL };
 #endif
 
 /// Treat the given \b val as a constant of \b size bytes
